Realm/Realm/Realm.cs

```diff
@@ -169,31 +169,30 @@ public static bool Compact(RealmConfigurationBase config = null)
         }
 
         /// <summary>
-        /// Deletes all the files associated with a realm.
+        /// Deletes all files associated with a given Realm if the Realm exists and is not open.
         /// </summary>
+        /// <remarks>
+        /// The Realm file must not be open on other threads.
+        /// This method must not be called inside a transaction.
+        /// All but the .lock file will be deleted by this.
+        /// </remarks>
         /// <param name="configuration">A <see cref="RealmConfigurationBase"/> which supplies the realm path.</param>
+        /// <exception cref="RealmInUseException">Thrown if the Realm is still open.</exception>
         public static void DeleteRealm(RealmConfigurationBase configuration)
         {
             Argument.NotNull(configuration, nameof(configuration));
 
-            var fullpath = configuration.DatabasePath;
-            if (IsRealmOpen(fullpath))
+            if (File.Exists(configuration.DatabasePath))
             {
-                throw new RealmPermissionDeniedException("Unable to delete Realm because it is still open.");
-            }
-
-            var filesToDelete = new[] { string.Empty, ".log_a", ".log_b", ".log", ".lock", ".note" }
-                .Select(ext => fullpath + ext)
-                .Where(File.Exists);
+                // Race condition:
+                // When trying to delete the Realm it can in some occasions (usually when Sync is involved)
+                // still be in use. To make sure other threads that use the same Realm get scheduled again
+                // and can finish their work before we actually delete the Realm files, we have to wait for
+                // a moment here.
+                // This can be removed as soon as https://github.com/realm/realm-core/issues/4762 is resolved.
+                Task.Delay(1).Wait();
```

Comment:
I had to add this to fix the bug where the lock was still in use. But with the extended wait in the test, it maybe is fixed anyway. I'll push a version without.

Reply:
Just double checked. I guess we confused ourselves.

The one in the test code (the base class tear down) was the one necessary for `GetInstance_WhenDynamicAndDoesntExist_ReturnsEmptySchema`.

This one here is the workaround for `DeleteRealmWorksIfCalledMultipleTimes` (also in `SynchronizedInstanceTests`):
https://github.com/realm/realm-core/issues/4762

Reply:
@nirinchev I've just re-added it. When you compare the two last commits it seems to make the difference, unfortunately. But since it's only 5ms and only on delete it seems to be ok as a workaround until the Core bug is fixed.